### DISCUSSION TITLE
Naming routes

### DISCUSSION THREAD

**Discussion by andre-dasilva** [tokio-rs/axum#2177] [Q&A] [answered]
Hi there, quick question. In actix you can set names for your routes

e.q
```rust
.service(
    web::resource("/about")
        .route(web::get().to(about))
        .name("about"),
)
```
that you can retrieve on the request with:
```rust
request.url_for_static(route_name)
```

I know such a feature also exists in like django (python).

The reason for it, is that i would like to give my routes a name and not hardcode the url everywhere. 
Is there something similar in axum?

**Comment by jplatte** [comment]
There is nothing similar in axum, but axum-extra provides the [`TypedPath`](https://docs.rs/axum-extra/latest/axum_extra/routing/trait.TypedPath.html) trait and derive macro which also derives a `Display` implementation for the same use case.

**Comment by davidpdrsn** [comment]
No axum doesn’t have such a feature. So you have to handle that yourself.

You can also use https://docs.rs/axum-extra/0.7.7/axum_extra/routing/trait.TypedPath.html from axum-extra

**Reply by andre-dasilva** [reply]
thanks. i will take a look at TypedPath. 
is there any plans to add such a feature or does TypedPath solve it?

**Reply by davidpdrsn** [reply, marked as answer]
There are no plans to add something like this to axum itself.

**Reply by andre-dasilva** [reply]
thanks everyone for the replies. i will take a look at TypedPath
